enhancements/observability/multicluster-observability-addon.md

```diff
@@ -0,0 +1,714 @@
+        namespace: open-cluster-management
+      spec:
+        config: |
+          receivers:
+            jaeger:
+              protocols:
+                grpc:
+            otlp:
+              protocols:
+                grpc:
+                http:
+          processors:
+          exporters:
+            debug:
+            otlp:
+              tls:
+                insecure: true
+          service:
+            pipelines:
+              traces:
+                receivers: [jaeger,otlp]
+                processors: []
+                exporters: [otlp, debug]
+```
+
+
+#### Hypershift [optional]
+
+TBD
+
+### Drawbacks
+
+TBD
+
+## Design Details
+
+### Open Questions [optional]
+
+The [addon-framework](ocm-addon-framework) is still lacking some features that will fully enable the behaviours described in this proposal namelly: 
+- [ ] Support for reconcile resources generated by the addon (Jira issues [LOG-5343](https://issues.redhat.com/browse/LOG-5343) and [TRACING-4148](https://issues.redhat.com/browse/TRACING-4148)). MCOA needs this otherwise it will not react on changes to those resources.
+- [ ] Support for multiple resources of the same kind in the ManagedClusterAddOn resource (Jira issues [LOG-5344](https://issues.redhat.com/browse/LOG-5344)). MCOA needs this otherwise it will not react on changes to those resources.
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+None
+
+### Upgrade / Downgrade Strategy
+
+None
+
+### Version Skew Strategy
+
+None
+
+### Operational Aspects of API Extensions
+
+TBD
+
+#### Failure Modes
+
+TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Alternatives
+
+TBD
+
+## Infrastructure Needed [optional]
+
+None
+
+[ocm-addon-framework]:https://github.com/open-cluster-management-io/addon-framework
+[ocp-cluster-logging-operator]:https://github.com/openshift/cluster-logging-operator
+[ocp-clusterlogforwarder-outputsecretspec]:https://github.com/openshift/cluster-logging-operator/blob/627b0c7f8c993f89250756d9601d1a632b024c94/apis/logging/v1/cluster_log_forwarder_types.go#L226-L265
+[ocp-clusterlogforward-outputtypespec]:https://github.com/openshift/cluster-logging-operator/blob/627b0c7f8c993f89250756d9601d1a632b024c94/apis/logging/v1/output_types.go#L21-L40
+[opentelemetry-collector-auth]:https://opentelemetry.io/docs/collector/configuration/#authentication
+[opentelemetry-operator]:https://github.com/open-telemetry/opentelemetry-operator
+[rhacm-multi-cluster-observability]:https://github.com/stolostron/multicluster-observability-operator
+
+## RANDOM  IDEAS 
+- What happens if we need to reference multiple static secrets?
+- Consolidate annotation/labels, namely authentication annotations can/should be uniform
+- Maybe we can use annotations for the authentication methods instead of having to create a seperate resouce for that
```

Comment:
> @periklis Here are some of the improvements I want to make to the EP, let me know your thoughts
> 
> > * What happens if we need to reference multiple static secrets?
> 
> Currently, if an output uses `StaticAuthentication` we look for a secret called `static-authentication` under `open-cluster-management` however this is not a correct assumption. A user might want to use different auth for a different clusterset. My idea would be that the users has to create the secret and add a label/annotation that will identify the secret. Then this secret will go on the list of resources in `ManagedClusterConfig`
> 

Yes that's probably a hindsight of my initial proposal writing that did not consider implications of per-clusterset configuration.

> > * Maybe we can use annotations for the authentication methods instead of having to create a separate resource for that
> 
> Currently, we require users to create a configmap with `output: authentication` however I think we can move this data to the annotations of the stanza. This way it's one less resource the user has to create and it help centralize the configuration. WDYT?
> 

SGTM.

> > * Consolidate annotation/labels, namely authentication annotations can/should be uniform
> 
> [rhobs/multicluster-observability-addon#44](https://github.com/rhobs/multicluster-observability-addon/pull/44)

SGTM. We already have too many configmaps/secrets to consider per `ClusterLogForwarder` and `OpenTelemetryCollector` stanza.